pkg/relayer/config/relayminer_configs_reader.go

```diff
@@ -4,98 +4,264 @@ import (
-	if yamlRelayMinerConfig.QueryNodeUrl == "" {
-		return nil, ErrRelayMinerConfigInvalidQueryNodeUrl.Wrapf("query node url is required")
+	// Top level section
+	// SigningKeyName is required
+	if yamlRelayMinerConfig.SigningKeyName == "" {
+		return nil, ErrRelayMinerConfigInvalidSigningKeyName
 	}
 
-	// Parse the query node URL
-	queryNodeUrl, err := url.Parse(yamlRelayMinerConfig.QueryNodeUrl)
-	if err != nil {
-		return nil, ErrRelayMinerConfigInvalidQueryNodeUrl.Wrapf("%s", err)
+	// SmtStorePath is required
+	if yamlRelayMinerConfig.SmtStorePath == "" {
+		return nil, ErrRelayMinerConfigInvalidSmtStorePath
 	}
 
-	// Check that the network node URL is provided
-	if yamlRelayMinerConfig.NetworkNodeUrl == "" {
-		return nil, ErrRelayMinerConfigInvalidNetworkNodeUrl.Wrapf("network node url is required")
+	// Pocket node urls section
+	relayMinerPocketConfig := &RelayMinerPocketConfig{}
+	pocket := yamlRelayMinerConfig.Pocket
+
+	if pocket.TxNodeGRPCUrl == "" {
+		return nil, ErrRelayMinerConfigInvalidNodeUrl.Wrap("tx node grpc url is required")
 	}
 
-	// Parse the network node URL
-	networkNodeUrl, err := url.Parse(yamlRelayMinerConfig.NetworkNodeUrl)
+	// Check if the pocket node grpc url is a valid URL
+	relayMinerPocketConfig.TxNodeGRPCUrl, err = url.Parse(pocket.TxNodeGRPCUrl)
 	if err != nil {
-		return nil, ErrRelayMinerConfigInvalidNetworkNodeUrl.Wrapf("%s", err)
+		return nil, ErrRelayMinerConfigInvalidNodeUrl.Wrapf(
+			"invalid tx node grpc url %s",
+			err.Error(),
+		)
 	}
 
-	// Parse the websocket URL of the Pocket Node to connect to for subscribing to on-chain events.
-	pocketNodeWebsocketUrl := sdk.HostToWebsocketURL(queryNodeUrl.Host)
+	// If the query node grpc url is empty, use the tx node grpc url
+	if pocket.QueryNodeGRPCUrl == "" {
+		relayMinerPocketConfig.QueryNodeGRPCUrl = relayMinerPocketConfig.TxNodeGRPCUrl
+	} else {
+		// If the query node grpc url is not empty, make sure it is a valid URL
+		relayMinerPocketConfig.QueryNodeGRPCUrl, err = url.Parse(pocket.QueryNodeGRPCUrl)
+		if err != nil {
+			return nil, ErrRelayMinerConfigInvalidNodeUrl.Wrapf(
+				"invalid query node grpc url %s",
+				err.Error(),
+			)
+		}
+	}
 
-	if yamlRelayMinerConfig.SigningKeyName == "" {
-		return nil, ErrRelayMinerConfigInvalidSigningKeyName
+	if pocket.QueryNodeRPCUrl == "" {
+		return nil, ErrRelayMinerConfigInvalidNodeUrl.Wrap("query node rpc url is required")
 	}
 
-	if yamlRelayMinerConfig.SmtStorePath == "" {
-		return nil, ErrRelayMinerConfigInvalidSmtStorePath
+	// Check if the query node rpc url is a valid URL
+	relayMinerPocketConfig.QueryNodeRPCUrl, err = url.Parse(pocket.QueryNodeRPCUrl)
+	if err != nil {
+		return nil, ErrRelayMinerConfigInvalidNodeUrl.Wrapf(
+			"invalid query node rpc url %s",
+			err.Error(),
+		)
 	}
 
-	if yamlRelayMinerConfig.ProxiedServiceEndpoints == nil {
-		return nil, ErrRelayMinerConfigInvalidServiceEndpoint.Wrapf("proxied service endpoints are required")
+	// Proxies section
```

Comment:
This might require a refactor, so want to hear your thoughts first. 

Rather than having comments for sections, what if you create small helper functions and have `RelayMinerConfig,` be the receiver? Then the code might look like:

```go
err := relayMinerConfig.HydrateProxies(yamlConfig)
if err != nil {...}
relayMinerConfig.HydrateSuppliers(yamlConfig)
if err != nil {...}
```